Mathlib/Algebra/Category/AlgebraCat/Basic.lean

```diff
@@ -25,6 +24,7 @@ variable (R : Type u) [CommRing R]
 
 /-- The category of R-algebras and their morphisms. -/
 structure AlgebraCat where
```

Comment:
You might be able to use
```suggestion
structure AlgebraCat where of ::
```
here, and then drop `of` below. Fine to leave to a follow-up PR though.

Reply:
Ah, this doesn't work because unfortunately `R` is implicit. Maybe @kmill knows if there's any syntax for overriding argument implicitness in constructors, or if any is planned.

Reply:
It saves having two identical functions (your current version means `mk ::`), which can be annoying if the wrong one ends up in your goal state.

Reply:
> Ah, this doesn't work because unfortunately `R` is implicit. Maybe @kmill knows if there's any syntax for overriding argument implicitness in constructors, or if any is planned.

This seems like it should be filed with core. There is clearly value in forcing the user to write `R`. We don't want Lean to initiate a search over all `R` that might `A` into an `Algebra R A`.

Reply:
At the least, `mk` should be `private` to avoid users introducing it and the doc-string of `of` should say something about it being the preferred constructor for `AlgebraCat`.

Reply:
How do I make the automatically generated constructor private?

Reply:
```lean
structure foo where 
  private mk :: 
```

Reply:
Thanks, done.